pkg/cpio/fs_unix.go

```diff
@@ -223,9 +223,39 @@ func (r *Recorder) inode(i Info) (Info, bool) {
 	return i, false
 }
 
+type null struct{}
+
+func (*null) ReadAt([]byte, int64) (int, error) {
+	return -1, io.EOF
+}
+
 func newLazyFile(name string) io.ReaderAt {
 	return uio.NewLazyOpenerAt(func() (io.ReaderAt, error) {
-		return os.Open(name)
+		f, oerr := os.Open(name)
+		if oerr == nil {
+			return f, oerr
+		}
+		// There are a few special case files that have
+		// only write permissions set. In those cases,
+		// create a record for a zero-length file.
+		if !os.IsPermission(oerr) {
+			return nil, oerr
+		}
+		i, err := os.Stat(name)
+		if err != nil {
+			return nil, oerr
+		}
+		// is it ONLY writeable?
+		m := i.Mode()
+		// If there are not-write modes set, then
+		// this is not just a writeonly file
+		if m&0555 != 0 {
+			return f, oerr
+		}
+		// At this point, it is either a file which
+		// only allows write or has no permissions at all.
+		// Either way, we can include it.
+		return &null{}, nil
```

Comment:
return nil, nil should work, too. I don't think null is necessary

Reply:
it's pretty necessary:
--- FAIL: TestWriteOnlyFile (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x51a2d7]

Reply:
Can you post the goroutine associated with that stack trace? 

Because it shouldn't be necessary, and if it is, we gotta fix that, too. https://github.com/u-root/u-root/blob/a0a3b250d3881425eb0a306885ae016664ea5a77/pkg/cpio/newc.go#L157

Reply:
--- FAIL: TestWriteOnlyFile (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x51a2d7]

goroutine 20 [running]:
testing.tRunner.func1(0xc0000c4300)
	/home/rminnich/go/src/testing/testing.go:827 +0x388
panic(0x54d940, 0x692ec0)
	/home/rminnich/go/src/runtime/panic.go:522 +0x1b5
github.com/u-root/u-root/pkg/uio.(*LazyOpenerAt).ReadAt(0xc0001bba70, 0xc0001c8000, 0x8000, 0x8000, 0x0, 0x69b920, 0x7fdbb2c046d0, 0x0)
	/home/rminnich/gopath/src/github.com/u-root/u-root/pkg/uio/lazy.go:88 +0x67
io.(*SectionReader).Read(0xc0001bbad0, 0xc0001c8000, 0x8000, 0x8000, 0x0, 0xc00006b600, 0x4097b8)
	/home/rminnich/go/src/io/io.go:475 +0x78
io.copyBuffer(0x5a6260, 0xc0001bba10, 0x5a62c0, 0xc0001bbad0, 0xc0001c8000, 0x8000, 0x8000, 0x0, 0x8, 0x2)
	/home/rminnich/go/src/io/io.go:402 +0x122
io.Copy(...)
	/home/rminnich/go/src/io/io.go:364
github.com/u-root/u-root/pkg/cpio.(*writer).WriteRecord(0xc0001bba10, 0x7fdbb099c168, 0xc0001bba70, 0x0, 0x8000, 0x3e8, 0x3e8, 0x1, 0x5d0d9214, 0x0, ...)
	/home/rminnich/gopath/src/github.com/u-root/u-root/pkg/cpio/newc.go:162 +0x4c4
github.com/u-root/u-root/pkg/cpio.(*DedupWriter).WriteRecord(0xc000092ce0, 0x7fdbb099c168, 0xc0001bba70, 0x0, 0x8000, 0x3e8, 0x3e8, 0x1, 0x5d0d9214, 0x0, ...)
	/home/rminnich/gopath/src/github.com/u-root/u-root/pkg/cpio/utils.go:127 +0x13c
github.com/u-root/u-root/pkg/cpio.TestWriteOnlyFile(0xc0000c4300)
	/home/rminnich/gopath/src/github.com/u-root/u-root/pkg/cpio/newc_test.go:174 +0x6f4
testing.tRunner(0xc0000c4300, 0x583688)
	/home/rminnich/go/src/testing/testing.go:862 +0xc0
created by testing.(*T).Run
	/home/rminnich/go/src/testing/testing.go:913 +0x357
exit status 2
FAIL	github.com/u-root/u-root/pkg/cpio	0.005s